src/plugins/postgres_table_copy_operator.py

```diff
@@ -112,6 +117,54 @@ def __init__(
                 "even though source table will be dropped."
             )
 
+    def _drop_tables_if_unequal(self, cursor: Any, all_table_copies: List[TableMapping]) -> None:
+        """Drop the target tables that are unequal to their source table."""
+        for table_mapping in all_table_copies:
+            # First make sure the target table exists
+            cursor.execute(
+                sql.SQL("SELECT to_regclass('{target_table_name}')").format(
+                    target_table_name=sql.Identifier(table_mapping.target)
+                )
+            )
+
+            if first(cursor.fetchone()) is None:
+                self.log.info(
+                    "Target table %s not found, skipping equality check.", table_mapping.target
+                )
+                continue
+            cursor.execute(
+                sql.SQL(
+                    """
+                    WITH src AS
+                      (SELECT data_type,
+                              ordinal_position,
+                              column_name
+                       FROM information_schema.columns
+                       WHERE table_name = %(source_table_name)s ),
+                         tgt AS
+                      (SELECT data_type,
+                              ordinal_position,
+                              column_name
+                       FROM information_schema.columns
+                       WHERE table_name = %(target_table_name)s )
+                    SELECT COUNT(*) = COUNT(src) AND count(*) = COUNT(tgt)
+                    FROM src NATURAL
+                    FULL OUTER JOIN tgt
+            """
+                ),
+                {
+                    "source_table_name": table_mapping.source,
+                    "target_table_name": table_mapping.target,
+                },
+            )
+            if not first(cursor.fetchone()):
+                self.log.info("Dropping table %s", table_mapping.target)
+                cursor.execute(
+                    sql.SQL("DROP TABLE {table_name}").format(
+                        table_name=sql.Identifier(table_mapping.target)
```

Comment:
Yes! :+1:  Here the table name is indeed an identifier from the SQL grammar point of view. Hence you need to use `postgresql.sql` helpers.